src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_EditProfileScreen.tsx

```diff
@@ -0,0 +1,324 @@
+import {CustomerProfile} from '@atb/api/types/profile';
+import {numberToAccessibilityString} from '@atb/utils/accessibility';
+
+type EditProfileScreenProps = ProfileScreenProps<'Profile_EditProfileScreen'>;
+type SubmissionStatus =
+  | 'loading'
+  | 'invalid-email'
+  | 'unavailable-email'
+  | 'submitted'
+  | 'submission-error';
+type ProfileState = 'loading' | 'success' | 'error';
+
+export const Profile_EditProfileScreen = ({
+  navigation,
+}: EditProfileScreenProps) => {
+  const styles = useStyles();
+  const {t} = useTranslation();
+  const {
+    authenticationType,
+    customerNumber,
+    phoneNumber: authPhoneNumber,
+  } = useAuthState();
+
+  const [customerProfile, setCustomerProfile] = useState<CustomerProfile>();
+  const [email, setEmail] = useState('');
+  const [firstName, setFirstName] = useState('');
+  const [surname, setSurname] = useState('');
+  const [submissionStatus, setSubmissionStatus] = useState<
+    SubmissionStatus | undefined
+  >(undefined);
+  const [profileState, setProfileState] = useState<ProfileState>();
+
+  const isLoadingOrSubmittingProfile =
+    submissionStatus === 'loading' || profileState === 'loading';
+
+  const phoneNumber = parsePhoneNumber(
+    authPhoneNumber ?? '',
+  )?.formatInternational();
+
+  const onSubmit = async () => {
+    setSubmissionStatus('loading');
+
+    if (isValidEmail(email)) {
+      const {available} = await emailAvailable(email);
+      if (available || customerProfile?.email === email) {
+        try {
+          await updateProfile({
+            firstName,
+            surname,
+            email,
+          });
+          await fetchProfile();
+          setSubmissionStatus('submitted');
+        } catch {
+          setSubmissionStatus('submission-error');
+        }
+      } else {
+        setSubmissionStatus('unavailable-email');
+      }
+    } else {
+      setSubmissionStatus('invalid-email');
+    }
+  };
+
+  const getEmailErrorText = (
+    status: SubmissionStatus | undefined,
+  ): string | undefined => {
+    switch (status) {
+      case 'invalid-email':
+        return t(EditProfileTexts.personalDetails.email.formattingError);
+      case 'unavailable-email':
+        return t(EditProfileTexts.personalDetails.email.unavailableError);
+    }
+  };
+
+  const prefillPersonalia = (
+    response: string,
+    set: React.Dispatch<React.SetStateAction<string>>,
+  ) => {
+    // Receiving "_" from Entur when firstName or surname are not set on user profile, instead setting them to ""
+    if (response === '_') {
+      set('');
+    } else if (response) {
+      set(response);
+    }
+  };
+
+  const fetchProfile = useCallback(async () => {
+    try {
+      const response = await getProfile();
+      setCustomerProfile(response);
+
+      prefillPersonalia(response.email, setEmail);
+      prefillPersonalia(response.firstName, setFirstName);
+      prefillPersonalia(response.surname, setSurname);
```

Comment:
```suggestion
      const nonUnderscoreString = (str: string) => (str === '_' ? '' : str);
      setEmail(nonUnderscoreString(response.email));
      setFirstName(nonUnderscoreString(response.firstName));
      setSurname(nonUnderscoreString(response.surname));
```
You can consider if you like this solution. It is a bit shorter because it allows `prefillPersonalia` to be removed.